web3/providers/ipc.py

```diff
@@ -42,22 +42,52 @@ def get_default_ipc_path(testnet=False):
         testnet = ""
 
     if sys.platform == 'darwin':
-        return os.path.expanduser(os.path.join(
+        ipc_path = os.path.expanduser(os.path.join(
             "~",
             "Library",
             "Ethereum",
             testnet,
-            "geth.ipc",
+            "geth.ipc"
         ))
+        if os.path.exists(ipc_path):
+            return ipc_path
+
+        ipc_path = os.path.expanduser(os.path.join(
+            "~",
+            "Library",
+            "Application Support",
+            "io.parity.ethereum",
+            "jsonrpc.ipc"
+        ))
+        if os.path.exists(ipc_path):
+            return ipc_path
+
     elif sys.platform.startswith('linux'):
-        return os.path.expanduser(os.path.join(
+        ipc_path = os.path.expanduser(os.path.join(
             "~",
             ".ethereum",
             testnet,
-            "geth.ipc",
+            "geth.ipc"
         ))
+        if os.path.exists(ipc_path):
+            return ipc_path
+
+        ipc_path = os.path.expanduser(os.path.join(
+            "~",
+            ".local",
+            "share",
+            "io.parity.ethereum",
+            "jsonrpc.ipc"
+        ))
+        if os.path.exists(ipc_path):
+            return ipc_path
+
     elif sys.platform == 'win32':
-        return "\\\\.\\pipe\\geth.ipc"
+        if os.path.exists("\\\\.\\pipe\\geth.ipc"):
```

Comment:
I think these should still use `os.path.join` to construct these paths as it will use the proper path separators for the os.

Reply:
Heh, I've been out of windows for so long that I don't remember what `\\.\pipe\geth.ipc` even means.

@Sebohe because of the '\\' start, it wasn't intuitive to me how to use join here. This was the first way I could find to do the join: `os.path.join('\\\\', '.', 'pipe', 'geth')`

It looks like this is the last change needed to get merged.